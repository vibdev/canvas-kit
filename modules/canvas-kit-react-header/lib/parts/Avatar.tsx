/**
 * This is a placeholder component until we get the details of an avatar ironed out.
 *
 * @alex.nicholls no need to review unless it's for show/tell
 */
import * as React from 'react';
import {HeaderTheme} from '../shared/types';
import {css} from 'emotion';
import {colors} from '@workday/canvas-kit-react-core';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {userIcon} from '@workday/canvas-system-icons-web';

export type AvatarProps = {
  themeColor?: HeaderTheme;
};

export const Avatar: React.SFC<AvatarProps> = ({themeColor = HeaderTheme.White}: AvatarProps) => {
  const avatarSize = '32px';
  const avatarStyle = css({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: avatarSize,
    width: avatarSize,
    backgroundColor: themeColor === HeaderTheme.White ? colors.soap300 : colors.blueberry400,
    borderRadius: '9999px',
    boxSizing: 'border-box',
  });
  const iconColor = themeColor === HeaderTheme.White ? colors.licorice500 : colors.frenchVanilla100;

  return (
    <div className={avatarStyle}>
      <SystemIcon icon={userIcon} color={iconColor} size={20} />
    </div>
  );
};
