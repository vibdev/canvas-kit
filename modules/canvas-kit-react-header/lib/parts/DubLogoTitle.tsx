import * as React from 'react';
import {css, cx} from 'emotion';
import {colors} from '@workday/canvas-kit-react-core';
import {HeaderTheme, HeaderThemePropType} from '@workday/canvas-kit-react-header/lib/shared/types';
import {
  logoTitleStyle,
  verticalCenterStyle,
} from '@workday/canvas-kit-react-header/lib/shared/styles';
import * as chroma from 'chroma-js';
import {dubLogoWhite, dubLogoBlue} from './_brand-assets';

export type DubTitleProps = {
  title: string;
  bgColor?: string;
} & Partial<HeaderThemePropType>;

export class DubLogoTitle extends React.Component<DubTitleProps> {
  static defaultProps = {
    theme: HeaderTheme.white,
  };

  render() {
    const logoStyle = css({
      padding: '0 16px',
      marginRight: '16px',
      borderRight: `1px solid ${
        this.props.theme === HeaderTheme.white
          ? colors.soap400
          : chroma(colors.frenchVanilla100)
              .alpha(0.3)
              .css()
      }`,
    });

    const titleColor = css({
      color: this.props.theme === HeaderTheme.white ? colors.blueberry500 : colors.frenchVanilla100,
    });

    const bgColor = css({
      background: this.props.bgColor ? this.props.bgColor : 'none',
    });

    return (
      <div className={cx(css({display: 'inline-block'}), bgColor)}>
        <div className={cx(verticalCenterStyle)}>
          <span
            className={logoStyle}
            dangerouslySetInnerHTML={{
              __html: this.props.theme === HeaderTheme.white ? dubLogoBlue : dubLogoWhite,
            }}
          />
          <h3 className={cx(logoTitleStyle, titleColor)}>{this.props.title}</h3>
        </div>
      </div>
    );
  }
}
