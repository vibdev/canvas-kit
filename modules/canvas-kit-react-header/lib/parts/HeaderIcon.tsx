import * as React from 'react';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {colors} from '@workday/canvas-kit-react-core';
import {HeaderTheme, HeaderThemePropType} from '@workday/canvas-kit-react-header/lib/shared/types';

export type HeaderIconProps = {
  icon: CanvasSystemIcon;
} & Partial<HeaderThemePropType>;

export class HeaderIcon extends React.Component<HeaderIconProps> {
  static defaultProps = {
    theme: HeaderTheme.white,
  };

  render() {
    const color =
      this.props.theme === HeaderTheme.white ? colors.licorice200 : colors.frenchVanilla100;
    const colorHover =
      this.props.theme === HeaderTheme.white ? colors.licorice500 : colors.frenchVanilla400;
    return <SystemIcon icon={this.props.icon} color={color} colorHover={colorHover} />;
  }
}
