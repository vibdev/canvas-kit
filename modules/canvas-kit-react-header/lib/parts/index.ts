export {DubLogoTitle} from './DubLogoTitle';
export {DubLogoProduct} from './DubLogoProduct';
export {WorkdayLogoTitle} from './WorkdayLogoTitle';
export {Avatar} from './Avatar';
