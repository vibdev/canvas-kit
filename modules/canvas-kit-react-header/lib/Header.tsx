import * as React from 'react';
import {css} from 'emotion';
import styled from 'react-emotion';
import {type, spacing} from '@workday/canvas-kit-react-core';
import {DubLogoTitle, Search, WorkdayLogoTitle} from './parts';
import {themes} from './shared/themes';
import {HeaderTheme, HeaderVariant, HeaderHeight} from './shared/types';
import {SystemIcon, SystemIconProps} from '@workday/canvas-kit-react-icon';
import {justifyIcon} from '@workday/canvas-system-icons-web';
import throttle from 'lodash-es/throttle';
import {makeMq} from '@workday/canvas-kit-react-common';

export interface HeaderProps {
  /**
   * A HeaderTheme enum indicating which theme to use (White, Blue or Transparent)
   */
  themeColor: HeaderTheme;
  /**
   * A HeaderVariant enum indicating whether to use the Dub (small) or Full (large) version
   */
  variant: HeaderVariant;
  /**
   * The title to display in the header. Not used if `brand` is provided
   */
  title?: string;
  /**
   * A React node that will replace the dub logo and title if provided.
   */
  brand?: React.ReactNode;
  /**
   * The URL href of the logo link
   */
  brandUrl?: string;
  /**
   * True if the nav should be centered. False if it should be aligned right
   */
  centeredNav?: boolean;
  /**
   * True if the search input should be left aligned with the logo. False if it should be aligned right
   */
  highlightSearch?: boolean;
  /**
   * An event handler function that gets called when the responsive menu icon is clicked
   */
  onMenuClick?: (e: React.SyntheticEvent) => void;
  /**
   * An event handler function that gets called when the search field is submitted
   */
  onSearchSubmit?: (query: string) => void;
  /**
   * An object that allows for custom specified breakpoints (sm, md, lg)
   */
  breakpoints: {
    [key: string]: number;
    sm: number;
    md: number;
    lg: number;
  };
}

export interface HeaderState {
  screenSize: keyof HeaderProps['breakpoints'];
}

const HeaderShell = styled('div')<HeaderProps>(
  {
    overflow: 'hidden',
    display: 'flex',
    alignItems: 'center',
    boxSizing: 'border-box',
    ...type.body,
    WebkitFontSmoothing: 'antialiased',
    MozOsxFontSmoothing: 'grayscale',
  },
  ({variant, themeColor}) => ({
    height: variant === HeaderVariant.Dub ? HeaderHeight.Small : HeaderHeight.Large,
    background: themes[themeColor].background,
    ...themes[themeColor].depth,
    color: themes[themeColor].color,
  })
);

const BrandSlot = styled('div')(
  {
    height: '100%',
  },
  (props: {grow?: boolean}) => ({
    flexGrow: props.grow ? 'unset' : 1,
  })
);

const BrandLink = styled('a')({
  ...type.link,

  '&:hover': {
    textDecoration: 'none',
  },

  '&:focus': {
    textDecoration: 'none',
  },
});

const navStyle = (props: HeaderProps) => {
  const theme = themes[props.themeColor];

  return css({
    nav: {
      display: 'flex',
      flexGrow: 1,
      justifyContent: 'center',
      height: 'inherit',

      '& ul': {
        color: theme.linkColor,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        listStyleType: 'none',
        padding: 0,
        margin: 0,
        height: 'inherit',
        '&:hover': {
          // De-emphasizes all color on hover, this allows us to create an illusion that hovering over a specific element
          // fades out the rest
          color: theme.linkFadeOutColor,
        },
        '& li': {
          boxSizing: 'border-box',
          position: 'relative',
          display: 'flex',
          alignItems: 'center',
          padding: `0px ${spacing.s}`,
          margin: `0 ${spacing.xxxs}`,
          fontSize: '14px',
          fontWeight: 700,
          height: 'inherit',
          transition: `color 150ms ease-out 0s`,
          cursor: 'pointer',
          '&:first-child': {
            marginLeft: 0,
          },
          '&:last-child': {
            marginRight: 0,
          },
          '& a': {
            color: 'inherit',
            textDecoration: 'none',
            '&:visited': {
              color: 'inherit', // Keeps visited links from becoming default purple
            },
          },
          '&:hover, &:active': {
            color: theme.linkColor, // Completes the illusion of sibling elements fading into the background on hover
          },
          '&.current': {
            color: theme.currentLinkColor,
            '&:hover, &:active': {
              color: theme.currentLinkColor,
            },
            '&:after': {
              position: 'absolute',
              bottom: 0,
              left: 0,
              content: `''`,
              height: 4,
              width: '100%',
              backgroundColor: theme.chipColor,
              borderRadius: '3px 3px 0 0',
            },
          },
        },
      },
    },
  });
};

const ChildrenSlot = styled('div')<HeaderProps>(({centeredNav = false, variant, breakpoints}) => {
  const mq = makeMq(breakpoints);

  return {
    marginRight: variant === HeaderVariant.Dub ? spacing.s : spacing.l,
    '> *': {
      marginRight: 0,
      marginLeft: spacing.s,
    },

    // TODO: remove this when we get real icon buttons
    '> .canvas-header--menu-icon': {
      cursor: 'pointer',
    },
    [mq.sm]: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      height: '100%',
      flexGrow: centeredNav ? 1 : 'unset',

      '> *:not(.canvas-header--menu-icon)': {
        display: 'none',
      },
    },
    [mq.md]: {
      '> *': {
        margin: `0 ${spacing.s}`,
      },

      '> *:last-child': {
        marginRight: 0,
      },

      '> *:not(.canvas-header--menu-icon)': {
        display: 'flex',
      },
    },
  };
},                                              navStyle);

class Brand extends React.Component<HeaderProps> {
  render() {
    return (
      <span>
        {this.props.variant === HeaderVariant.Dub &&
          (this.props.brand || (
            <DubLogoTitle
              title={this.props.title ? this.props.title : ''}
              themeColor={this.props.themeColor}
            />
          ))}
        {this.props.variant === HeaderVariant.Full &&
          (this.props.brand || (
            <WorkdayLogoTitle
              title={this.props.title ? this.props.title : ''}
              themeColor={this.props.themeColor}
            />
          ))}
      </span>
    );
  }
}

export default class Header extends React.Component<HeaderProps, HeaderState> {
  static Theme = HeaderTheme;
  static Variant = HeaderVariant;
  static defaultProps = {
    themeColor: HeaderTheme.White,
    variant: HeaderVariant.Dub,
    breakpoints: {
      sm: 320,
      md: 768,
      lg: 1120,
    },
  };

  constructor(props: HeaderProps) {
    super(props);
    this.state = {
      screenSize: this.getScreenSize(window.innerWidth, props.breakpoints),
    };
    this.updateScreenSize = throttle(this.updateScreenSize.bind(this), 150);
  }

  /**
   * Returns which breakpoint size to use (sm, md, lg)
   * @param {number} width Current screen width
   * @param {Object} breakpoints Specification of breakpoint pixel values for sm, md, lg screens
   * @returns {string}
   */
  getScreenSize(width: number, breakpoints: HeaderProps['breakpoints']): string {
    const sizes = Object.keys(breakpoints);
    let screenSize = sizes[0];

    sizes.forEach(k => {
      if (width >= breakpoints[k]) {
        screenSize = k;
      }
    });

    return screenSize;
  }

  /**
   * Updates the state screen size/width if it has changed (syncs w/ browser paints using RAF)
   */
  updateScreenSize(): void {
    requestAnimationFrame(() => {
      const currentScreenSize = this.getScreenSize(window.innerWidth, this.props.breakpoints);

      if (currentScreenSize !== this.state.screenSize) {
        this.setState({
          screenSize: currentScreenSize,
        });
      }
    });
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateScreenSize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateScreenSize);
  }

  /**
   * Helper that recursively maps ReactNodes to their theme-based equivalent.
   * Any children that are included in a Header may need to undergo transformation
   * before stamped out in render().
   *
   * E.g. <SystemIcon> components need to have the appropriate `color` and `colorHover`
   * props set based on the theme.cu
   *
   * @param children From props.children of a React component
   *
   * @returns {React.ReactNode} The child/children to be rendered
   */
  private renderChildren(children: React.ReactNode): React.ReactNode {
    return React.Children.map(children, child => {
      if (!React.isValidElement(child)) {
        return child;
      }

      // child is now guaranteed to be a valid ReactElement from the check above

      type Props = {children: React.ReactNode};
      const propsChildren = (child as React.ReactElement<Props>).props.children;

      if (React.Children.count(propsChildren)) {
        if (child.type === 'nav' && this.state.screenSize !== 'lg') {
          // Screen size is smaller than our largest breakpoint so turn nav into a hamburger
          const hamburgerIconProps = {
            color: themes[this.props.themeColor].systemIcon.color,
            colorHover: themes[this.props.themeColor].systemIcon.colorHover,
            icon: justifyIcon,
            onClick: this.props.onMenuClick,
          };

          // TODO: This needs to get changed to IconButton when we get it restyled for headers
          return (
            <SystemIcon {...hamburgerIconProps} className="canvas-header--menu-icon" tabIndex={0} />
          );
        }
        return React.cloneElement(child as React.ReactElement<Props>, {
          children: this.renderChildren(propsChildren),
        });
      }

      if (child.type === SystemIcon) {
        return React.cloneElement(child as React.ReactElement<SystemIconProps>, {
          color: themes[this.props.themeColor].systemIcon.color,
          colorHover: themes[this.props.themeColor].systemIcon.colorHover,
        });
      }

      return child;
    });
  }

  render() {
    return (
      <HeaderShell {...this.props}>
        <BrandSlot grow={this.props.centeredNav}>
          {this.props.brandUrl ? (
            <BrandLink href={this.props.brandUrl}>
              <Brand {...this.props} />
            </BrandLink>
          ) : (
            <Brand {...this.props} />
          )}
        </BrandSlot>
        {this.props.onSearchSubmit && (
          <Search
            onSubmit={this.props.onSearchSubmit}
            highlight={this.props.highlightSearch}
            themeColor={this.props.themeColor}
            placeholder="Search"
          />
        )}
        <ChildrenSlot {...this.props}>{this.renderChildren(this.props.children)}</ChildrenSlot>
      </HeaderShell>
    );
  }
}
