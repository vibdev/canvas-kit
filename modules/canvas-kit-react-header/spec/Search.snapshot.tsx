import * as React from 'react';
// import {mount} from 'enzyme';
import * as renderer from 'react-test-renderer';
import Header from '../lib/Header';
import {Search} from '../lib/parts/Search';

describe('Header Search Snapshots', () => {
  const cb = jest.fn();
  afterEach(() => {
    cb.mockReset();
  });

  // test('Renders a collapsed search bar', () => {
  //   const component = mount(<Search collapse={true} />);

  //   expect(component).toMatchSnapshot();
  // });

  test('renders themed search bars', () => {
    const tree = {
      blue: renderer.create(<Search themeColor={Header.Theme.Blue} />).toJSON(),
      transparent: renderer.create(<Search themeColor={Header.Theme.Transparent} />).toJSON(),
      white: renderer.create(<Search themeColor={Header.Theme.White} />).toJSON(),
    };
    expect(tree.blue).toMatchSnapshot('<Search /> with blue theme applied');
    expect(tree.transparent).toMatchSnapshot('<Search /> with transparent theme applied');
    expect(tree.white).toMatchSnapshot('<Search /> with white theme applied');
  });

  console.log('hi');

  // test('renders themed, collapsed search bars', () => {
  //   const tree = {
  //     blue: renderer.create(<Search themeColor={Header.Theme.Blue} collapse={true} />).toJSON(),
  //     transparent: renderer
  //       .create(<Search themeColor={Header.Theme.Transparent} collapse={true} />)
  //       .toJSON(),
  //     white: renderer.create(<Search themeColor={Header.Theme.White} collapse={true} />).toJSON(),
  //   };
  //   expect(tree.blue).toMatchSnapshot('<Search /> with blue theme applied');
  //   expect(tree.transparent).toMatchSnapshot('<Search /> with transparent theme applied');
  //   expect(tree.white).toMatchSnapshot('<Search /> with white theme applied');
  // });
});
