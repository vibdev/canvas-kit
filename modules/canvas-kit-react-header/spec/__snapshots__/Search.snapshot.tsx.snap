// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Search Snapshots Renders a collapsed search bar 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
    }
  }
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginLeft": "16px",
    }
  }
/>
`;

exports[`Header Search Snapshots renders themed search bars: <Search /> with blue theme applied 1`] = `
.emotion-3 {
  position: relative;
  margin-left: 24px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: block;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  padding: 12px;
  padding-left: 44px;
  max-width: 320px;
  min-width: 120px;
  width: 100%;
  height: 40px;
  border-radius: 3px;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: rgba(0,0,0,0.2);
  color: #ffffff;
}

.emotion-1::-webkit-input-placeholder {
  color: #ffffff;
}

.emotion-1::-moz-placeholder {
  color: #ffffff;
}

.emotion-1:-ms-input-placeholder {
  color: #ffffff;
}

.emotion-1::placeholder {
  color: #ffffff;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  outline: none;
  background: #ffffff;
  color: #494949;
}

.emotion-1:not([disabled]):focus::-webkit-input-placeholder,
.emotion-1:not([disabled]):active::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::-moz-placeholder,
.emotion-1:not([disabled]):active::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus:-ms-input-placeholder,
.emotion-1:not([disabled]):active:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::placeholder,
.emotion-1:not([disabled]):active::placeholder {
  color: #5e6a75;
}

<form
  className="emotion-3 emotion-4"
  onSubmit={[Function]}
  placeholder="Search"
>
  <span
    className="emotion-0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
    style={
      Object {
        "left": "12px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }
  />
  <input
    className="emotion-1 emotion-2"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    type="search"
  />
</form>
`;

exports[`Header Search Snapshots renders themed search bars: <Search /> with transparent theme applied 1`] = `
.emotion-3 {
  position: relative;
  margin-left: 24px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: block;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  padding: 12px;
  padding-left: 44px;
  max-width: 320px;
  min-width: 120px;
  width: 100%;
  height: 40px;
  border-radius: 3px;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: rgba(0,0,0,0.2);
  color: #ffffff;
}

.emotion-1::-webkit-input-placeholder {
  color: #ffffff;
}

.emotion-1::-moz-placeholder {
  color: #ffffff;
}

.emotion-1:-ms-input-placeholder {
  color: #ffffff;
}

.emotion-1::placeholder {
  color: #ffffff;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  outline: none;
  background: #ffffff;
  color: #494949;
}

.emotion-1:not([disabled]):focus::-webkit-input-placeholder,
.emotion-1:not([disabled]):active::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::-moz-placeholder,
.emotion-1:not([disabled]):active::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus:-ms-input-placeholder,
.emotion-1:not([disabled]):active:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::placeholder,
.emotion-1:not([disabled]):active::placeholder {
  color: #5e6a75;
}

<form
  className="emotion-3 emotion-4"
  onSubmit={[Function]}
  placeholder="Search"
>
  <span
    className="emotion-0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
    style={
      Object {
        "left": "12px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }
  />
  <input
    className="emotion-1 emotion-2"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    type="search"
  />
</form>
`;

exports[`Header Search Snapshots renders themed search bars: <Search /> with white theme applied 1`] = `
.emotion-3 {
  position: relative;
  margin-left: 24px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: block;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  padding: 12px;
  padding-left: 44px;
  max-width: 320px;
  min-width: 120px;
  width: 100%;
  height: 40px;
  border-radius: 3px;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: #f0f1f2;
  color: #494949;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  outline: none;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
  background: #f0f1f2;
  color: #494949;
}

.emotion-1:not([disabled]):focus::-webkit-input-placeholder,
.emotion-1:not([disabled]):active::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::-moz-placeholder,
.emotion-1:not([disabled]):active::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus:-ms-input-placeholder,
.emotion-1:not([disabled]):active:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1:not([disabled]):focus::placeholder,
.emotion-1:not([disabled]):active::placeholder {
  color: #5e6a75;
}

<form
  className="emotion-3 emotion-4"
  onSubmit={[Function]}
  placeholder="Search"
>
  <span
    className="emotion-0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
    style={
      Object {
        "left": "12px",
        "position": "absolute",
        "top": "50%",
        "transform": "translateY(-50%)",
      }
    }
  />
  <input
    className="emotion-1 emotion-2"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    type="search"
  />
</form>
`;

exports[`Header Search Snapshots renders themed, collapsed search bars: <Search /> with blue theme applied 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #d7eafc;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #d7eafc;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
    }
  }
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginLeft": "16px",
    }
  }
/>
`;

exports[`Header Search Snapshots renders themed, collapsed search bars: <Search /> with transparent theme applied 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #d7eafc;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #d7eafc;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
    }
  }
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginLeft": "16px",
    }
  }
/>
`;

exports[`Header Search Snapshots renders themed, collapsed search bars: <Search /> with white theme applied 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-search wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.32 15.906l2.859 2.858a.5.5 0 0 1 .003.717l-.7.7a.5.5 0 0 1-.718-.002l-2.858-2.86a8 8 0 1 1 1.414-1.414zM11 17a6 6 0 1 0 0-12 6 6 0 0 0 0 12z\\" class=\\"wd-icon-fill\\"/></g></svg>",
    }
  }
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginLeft": "16px",
    }
  }
/>
`;
